packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,274 @@
+                            taskId: task.id,
+                            osProcessId: proc.pid,
+                            terminalId: (options.processType === 'terminal') ? proc.id : undefined,
+                            processId: (options.processType === 'raw') ? proc.id : undefined,
+                            command: cmd,
+                            label: options.label,
+                            ctx: ctx === undefined ? '' : ctx
+                        };
+                        this.fireTaskCreatedEvent(newTask);
+                        resolve(newTask);
+
+                    } catch (error) {
+                        this.logger.error(`Error occured while creating task: ${error}`);
+                        reject(new Error(error));
+                    }
+                })
+                .catch(err => {
+                    reject(new Error(err));
+                });
+
+        });
+    }
+
+    protected fireTaskExitedEvent(event: ITaskExitedEvent) {
+        this.logger.debug(log => log(`task has exited:`, event));
+        // notify all clients. They can filter on context to see if it's relevant to them
+        this.clients.forEach(client => {
+            client.onTaskExit(event);
+        });
+    }
+
+    protected fireTaskCreatedEvent(event: ITaskInfo) {
+        this.logger.debug(log => log(`task created:`, event));
+        // notify all clients. They can filter on context to see if it's relevant to them
+        this.clients.forEach(client => {
+            client.onTaskCreated(event);
+        });
+    }
+
+    kill(id: number): Promise<void> {
+        const taskToKill = this.taskManager.get(id);
+        if (taskToKill !== undefined) {
+            taskToKill.kill();
+            this.logger.info(`Attempting to kill task id ${id}`);
+        }
+        return Promise.resolve();
+    }
+
+    /** Set the client we send the notifications-to. We expect that there might be
+     * multiple clients, so add them to a list as they are set
+     */
+    setClient(client: ITaskClient) {
+        this.clients.push(client);
+    }
+
+    /**
+     * uses heuristics to look-for a command. Will look into the system path, if the command
+     * is given without a path. Will resolve if a potential match is found, else reject. There
+     * is no garantee that a command we find will be the one executed, if multiple commands with
+     * the same name exist.
+     * @param command command name to look-for
+     * @param cwd current working directory
+     */
+    private async findCommand(command: string, cwd: string): Promise<string> {
+        const systemPath = process.env.PATH;
+
+        return new Promise<string>(async (resolve, reject) => {
+
+            if (path.isAbsolute(command)) {
+                if (await this.fileExists(command)) {
+                    resolve(command);
+                    return;
+                } else {
+                    const uri = new URI(cwd).resolve(command);
```

Comment:
Not sure I get this it's absolute, and it doesn't exist but we will check for cwd/absolutepath ?

Reply:
Indeed, this looks like dead code; if a command with an absolute path is not found, there's not much more we can do. Removed.